code/controllers/subsystems/initialization/map_finalization.dm

```diff
@@ -32,3 +35,48 @@
 		all_areas += A
 
 	sortTim(all_areas, /proc/cmp_name_asc)
+
+/proc/place_dungeon_spawns()
+	var/map_directory = "maps/dungeon_spawns/"
+	var/list/files = flist(map_directory)
+	var/start_time = world.time
+	var/dungeons_placed = 0
+	var/static/dmm_suite/maploader = new
+
+	var/dungeon_chance = config.dungeon_chance
+
+	log_ss("map_finalization","Attempting to create asteroid dungeons for [length(asteroid_spawn)] different areas, with [length(files) - 1] possible dungeons, with a [dungeon_chance]% chance to spawn a dungeon per area.")
+
+	var/always_prob = 0
+
+	for(var/turf/spawn_location in asteroid_spawn)
+
+		if(length(files) <= 0) //Sanity
+			log_ss("map_finalization","There aren't enough dungeon map files to fill the entire dungeon map. There may be less dungeons than expected.")
+			break
+
+		if(always_prob || prob(dungeon_chance))
+
+			var/chosen_dungeon = pick(files)
+			always_prob = 0
+
+			if(!dd_hassuffix(chosen_dungeon,".dmm")) //Don't read anything that isn't a map file
+				files -= chosen_dungeon
+				always_prob = 1 //We wasted our roll on a readme file so be sure that the next dungeon will contain an actual dungeon.
```

Comment:
Could selecting a readme file be prevented by making some sort of subdirectory next to the readme for all the map files? Unless that would cause some issues, it seems a bit odd to have readme file clash with the process.